plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceUtils.java

```diff
@@ -171,6 +172,12 @@ public static DBPDataSourceContainer getDataSourceBySpec(
                 case PARAM_AUTO_COMMIT:
                     autoCommit = CommonUtils.toBoolean(paramValue);
                     break;
+                case PARAM_CREATE:
+                    createNewDataSource = CommonUtils.toBoolean(paramValue);
+                    if (parameterHandler != null) {
+                        parameterHandler.setParameter(paramName, paramValue);
+                    }
```

Comment:
What does it do? I don't see such logic for the parameters above.

Reply:
It's one of three parameters which is stored in `InstanceConnectionParameters implements GeneralUtils.IParameterHandler`. And this parameter is not the same as others in non-default cases. It used to be handled in default branch and set to `parameterHandler` in the last `else` branch before fix.